File: GenerateSha256Base64Hex/GenerateSha256Base64Hex.java
package Generate_SHA256_Base64_Hex;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

public class GenerateSha256Base64Hex {
    public static void main(String[] args) throws NoSuchAlgorithmException {
        System.out.println("Generate a SHA-256 hash, Base64 en- and decoding and hex conversions");

        String plaintext = "The quick brown fox jumps over the lazy dog";
        System.out.println("plaintext:                " + plaintext);

        byte[] sha256Value = calculateSha256(plaintext);
        System.out.println("sha256Value (hex) length: " + sha256Value.length + " data: " + bytesToHex(sha256Value));

        String sha256Base64 = base64Encoding(sha256Value);
        System.out.println("sha256Value (base64):     " + sha256Base64);

        byte[] sha256ValueDecoded = base64Decoding(sha256Base64);
        System.out.println("sha256Base64 decoded to a byte array:");
        System.out.println("sha256Value (hex) length: " + sha256ValueDecoded.length + " data: " + bytesToHex(sha256ValueDecoded));

        String sha256HexString = "d7a8fbb307d7809469ca9abcb0082e4f8d5651e46d3cdb762d02d0bf37c9e592";
        byte[] sha256Hex = hexStringToByteArray(sha256HexString);
        System.out.println("sha256HexString converted to a byte array:");
        System.out.println("sha256Value (hex) length: " + sha256Hex.length + " data: " + bytesToHex(sha256Hex));
    }

    private static byte[] calculateSha256 (String string) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        return md.digest(string.getBytes(StandardCharsets.UTF_8));
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuffer result = new StringBuffer();
        for (byte b : bytes) result.append(Integer.toString((b & 0xff) + 0x100, 16).substring(1));
        return result.toString();
    }

    public static byte[] hexStringToByteArray(String s) {
        int len = s.length();
        byte[] data = new byte[len / 2];
        for (int i = 0; i < len; i += 2) {
            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
                    + Character.digit(s.charAt(i + 1), 16));
        }
        return data;
    }

    private static String base64Encoding(byte[] input) {
        return Base64.getEncoder().encodeToString(input);
    }
    private static byte[] base64Decoding(String input) {
        return Base64.getDecoder().decode(input);
    }
}

File: GenerateSha256Base64Hex/GenerateSha256Base64Hex.go
// https://replit.com/@javacrypto/CpcGoGenerateSha256Base64Hex#main.go
package main

import (
  "crypto/sha256"
  "encoding/hex"
  "encoding/base64"
	"fmt"
  "strconv"
)

func main() {

  fmt.Println("Generate a SHA-256 hash, Base64 en- and decoding and hex conversions")

  plaintext := string("The quick brown fox jumps over the lazy dog")
  fmt.Println("plaintext: " + plaintext)

  sha256Value := CalculateSha256(plaintext)
  sha256Length := len(sha256Value)
  fmt.Println("sha256Value (hex) length: " + strconv.Itoa(sha256Length) + " data: " + BytesToHex(sha256Value))

  sha256Base64 := Base64Encoding(sha256Value)
  fmt.Println("sha256Value (base64) " + sha256Base64)

  sha256ValueDecoded := Base64Decoding(sha256Base64)
  sha256ValueDecodedLength := len(sha256ValueDecoded)
  fmt.Println("sha256Base64 decoded to a byte array:")
  fmt.Println("sha256Value (hex) length: " + strconv.Itoa(sha256ValueDecodedLength) + " data: " + BytesToHex(sha256ValueDecoded))

  sha256HexString := "d7a8fbb307d7809469ca9abcb0082e4f8d5651e46d3cdb762d02d0bf37c9e592"
  sha256Hex := HexStringToByteArray(sha256HexString)
  sha256HexLength := len(sha256Hex)
  fmt.Println("sha256HexString converted to a byte array:")
  fmt.Println("sha256Value (hex) length: " + strconv.Itoa(sha256HexLength) + " data: " + BytesToHex(sha256Hex))
}

func CalculateSha256(dataToHash string)([]byte){
  msgHash := sha256.New()
  _, err := msgHash.Write([]byte(dataToHash))
  if err != nil {
	  panic(err)
  }
  msgHashSum := msgHash.Sum(nil)
  return msgHashSum
}

func BytesToHex(input []byte)(string){
  return hex.EncodeToString(input)
}

func HexStringToByteArray(input string)([]byte){
  data, err := hex.DecodeString(input)
  if err != nil {
    panic(err)
  }
  return data;
}

func Base64Encoding(input []byte)(string) {
  return base64.StdEncoding.EncodeToString(input)
}

func Base64Decoding(input string)([]byte) {
  data, err := base64.StdEncoding.DecodeString(input)
	if err != nil {
		return data
	}
  return data
}
